### DISCUSSION TITLE
Duvida qual modelo macbook escolher

### DISCUSSION THREAD

**Discussion by bru95** [Flutterando/forum#50] [Perguntas e Respostas]
Bom dia pessoal, estou com uma duvida cruel.

Para programar qual vocês modelo abaixo vcs acham melhor ? Vantagens e desvantagens.
1. MacBook Pro 2022 Apple M2 / Memória 8GB / SSD 256GB / 13.3"
2. MacBook Pro 2020 Apple M1 / Memória 8GB / SSD 512GB / 13.3"
3. MacBook Pro 2020 Intel Core i5 2.0GHz / Memória 16GB / SSD 512GB / 13.3"

Desde já agradeço quem puder me ajudar.

**Comment by FelipeAngeli** [comment]
Eu utilizo o MacBook Pro 2020 Apple M1 / Memória 8GB / SSD 250GB / 13.3", e estou super satisfeito com ele.

**Comment by bru95** [comment]
Atualmente tenho um Macbook Pro 2019, intel core i5, 8GB RAM e 256GB SSD. Mas to precisando de mais espaço, por isso estou vendendo para comprar um com mais armazenamento. 
Estou na duvida na escolha Apple Ship ou a RAM de 16GB. Atualmente esta tranquilo a questão de compatibilidade com os processadores da apple então?
Desenvolvo android nativo (java), flutter, php (laravel), postgres, ...

**Comment by eduardoflorence** [comment]
Eu tenho o MacBook Pro 2020 Apple M1 / Memória 8GB / SSD 256GB / 13.3"
Ele é muito bom, mas me arrependo de não ter comprado o de 16GB de memória. Ele é muito bom, mas quando temos muitas coisas abertas (emulador, etc) a memória fica no limite de 8GB e reinicio para dar uma melhorada.
Quando ao processador, o M1 e M2 são muito superiores ao Core i5.
Eu ficaria com sua opção 1, mas aumenta a memória para 16GB. Só para enfatizar, você tem que fazer isso no momento da compra, pois o macbook com M1/2 não permite upgrade de memória.

**Comment by bru95** [comment]
Certo, obrigada pelas respostas.
Infelizmente, não está dentro do meu orçamento M2 com 16GB de RAM. Por isso coloquei essas 3 opções, a dúvida seria entre uma memória RAM maior ou um processador melhor.

**Comment by DEVigodE** [comment]
Concordo plenamente com o @eduardoflorence, a questão não e o processar, pois o OS da apple e bem otimizada. Agora para programar os 16gb de ram são essenciais.

**Comment by bru95** [comment]
entendi, obrigada pessoal!

**Reply by caio-deiro** [reply]
Como o @ZIBMBRAZIL disse acima, menos dee 16gb voce vai sofrer um pouco com lentidoes ao debugar algo mais pesado, alem do proprio emulador consumir muita ram. Eu ja utilizei de 8 e hoje o de 16, e a diferenca é absurda. Se puder, opte tambem pelo SSD M2. o PC fica mais rapido e eficiente, mesmo em comparacao ao ssd SATA.

**Reply by bru95** [reply]
certo, então acho que vou ir de opção 3 mesmo.
